### PR TITLE
Add more types

### PR DESCRIPTION
**Pull request by cdamian**
Adding types that we are currently missing from some of the pallets.

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ version: '2'
 services:
   substrate:
     container_name: substrate
-    image: parity/substrate:3.0.0-dev-1b646b2
+    image: parity/polkadot:v0.9.17
```
Comment (mikiquantum) [inline docker-compose.yml:6]: we can prob use v0.9.21

File: types/address_test.go [hunk 2]
```diff
@@ -40,9 +45,26 @@ func TestChecksum(t *testing.T) {
 	assert.Equal(t, ss58d[len(ss58d)-2:], res[:2]) // Verified checksum
 }
 
+var (
+	addressFuzzOpts = []fuzzOpt{
```
Comment (mikiquantum) [inline types/address_test.go:49]: Such a nice addition!

File: types/test/README.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Types Test
```
Comment (mikiquantum) [inline types/test/README.md:1]: I think I need another session to understand this well. Looks very promising to quickly verify for standard chains if types can be decoded 👍

**Review by mikiquantum — APPROVED** [review]
Overall looks great! Would be cool to:
- Highlight backwards incompatible changes with previous versions, like public functions that have changed and so on
- Have more details around the test-gen for type parsing verification

Thanks!

**Comment by cdamian** [comment]
> Overall looks great! Would be cool to:
> 
>     * Highlight backwards incompatible changes with previous versions, like public functions that have changed and so on
> 
>     * Have more details around the test-gen for type parsing verification
> 
> 
> Thanks!

I don't think we have any backwards incompatible changes. I will definitely improve the documentation around the test data generator,

File: types/test/README.md (hunk 3 above)
Comment (cdamian) [reply types/test/README.md:1]: I've updated these docs, PTAL.

**Pull request merged**